src/opendr/perception/object_detection_2d/datasets/wider_face.py

```diff
@@ -0,0 +1,130 @@
+                        for bbox in bboxes:
+                            bbox = bbox.split(' ')
+                            # check invalid
+                            if int(bbox[7]) == 1:
+                                continue
+                            bbox_odr = BoundingBox(name=cls_id, left=float(bbox[0]), top=float(bbox[1]),
+                                                   width=float(bbox[2]), height=float(bbox[3]))
+                            if ignore_tiny and max(bbox_odr.width, bbox_odr.height) < 10:
+                                continue
+                            bboxes_list.append(bbox_odr)
+                        cur_line += 2 + n_faces
+                        if bboxes_list:
+                            self.bboxes.append(BoundingBoxList(boxes=bboxes_list))
+                            image_dir = os.path.join(self.root, 'WIDER_{}'.format(split), 'images')
+                            image_paths.append(os.path.join(image_dir, image_name))
+            else:
+                # test split, only images are available
+                image_dir = os.path.join(self.root, 'WIDER_{}'.format(split), 'images')
+                subdirs_names = [subdir for subdir in os.listdir(image_dir) if os.path.isdir(os.path.join(image_dir, subdir))]
+                print(subdirs_names)
+                image_paths = []
+                for subdir in subdirs_names:
+                    image_names = os.listdir(os.path.join(image_dir, subdir))
+                    image_paths.extend([os.path.join(image_dir, subdir, image_name) for image_name in image_names])
+                self.bboxes = [np.empty((1, 5)) for _ in range(len(image_paths))]
+
+        dataset_type = 'wider_face'
+        super().__init__(classes=classes, dataset_type=dataset_type, image_paths=image_paths,
+                         image_transform=image_transform, target_transform=target_transform,
+                         transform=transform, splits=splits, root=root)
+
+    def __getitem__(self, item):
+        image_path = self.image_paths[item]
+        label = self.bboxes[item]
+
+        img_np = cv2.imread(image_path)
+        img = Image(img_np)
+
+        if self._image_transform is not None:
+            img = self._image_transform(img)
+
+        if self._target_transform is not None:
+            label = self._target_transform(label)
+
+        if self._transform is not None:
+            return self._transform(img, label)
+        return img, label
+
+    def pull_image(self, item):
+        image_path = self.image_paths[item]
+        img = cv2.imread(image_path)
+        return img
+
+    def get_bboxes(self, item):
+        return self.bboxes[item]
+
+    def __len__(self):
+        return len(self.image_paths)
+
+
+if __name__ == '__main__':
+    from opendr.perception.object_detection_2d.utils.vis_utils import draw_bounding_boxes
+
+    dataset = WiderFaceDataset('/home/administrator/data/wider', splits=['train', 'val'])
+
+    print(len(dataset))
```

Comment:
```suggestion
```

Reply:
Does it serve a purpose or it's just a forgotten debug message? It should either be removed or print additional info.